google/cloud/opentelemetry/internal/recordable.cc

```diff
@@ -148,8 +150,28 @@ google::devtools::cloudtrace::v2::Span&& Recordable::as_proto() && {
 }
 
 void Recordable::SetIdentity(
-    opentelemetry::trace::SpanContext const& /*span_context*/,
-    opentelemetry::trace::SpanId /*parent_span_id*/) noexcept {}
+    opentelemetry::trace::SpanContext const& span_context,
+    opentelemetry::trace::SpanId parent_span_id) noexcept {
```

Comment:
I'm not sure how you can assert `noexcept` when you do things like construct strings.

Reply:
You can't.  These are virtual functions and we need to respect the specification.

I [tried (and failed)](https://github.com/open-telemetry/opentelemetry-cpp/issues/4#issuecomment-544949758) to convince these folks to drop the `noexcept`.

It seems that we will need to capture all exceptions.  Maybe something like this (


```cc
// in some header...
bool NoExceptAction(absl::FunctionRef<void()> action) noexcept;

// in some cc
#if GOOGLE_CLOUD_CPP_HAVE_EXCEPTIONS
bool NoExceptAction(absl::FunctionRef<void()> action) noexcept try {
  action();
  return true;
} catch(...) {
  return false;
}
#else
bool NoExceptAction(absl::FunctionRef<void()> action) noexcept {
  action(); // Will crash, would have crashed anyway.
  return true;
}
#endif

// In Recordable.cc
void Recordable::SetIdentity(...) {
  usable_ = usable_ & NoExceptAction(SetIdentityImpl(...));
}
```

Then we will drop (not send to Cloud Trace) any `Recordable`s where `usable_` is `false`.

This is all for a TODO bug and future PRs.

Reply:
Opened #11288